morpheus/io/data_manager.py

```diff
@@ -0,0 +1,217 @@
+
+import cudf
+
+from morpheus.io.data_record import DataRecord
+
+
+class DataManager():
+    """
+    DataManager class to manage the storage and retrieval of files
+    using either in-memory or filesystem storage.
+    """
+
+    VALID_STORAGE_TYPES = ('in_memory', 'filesystem')
+    VALID_FILE_FORMATS = ('parquet', 'csv')
+
+    def __init__(self, storage_type: str = 'in_memory', file_format: str = 'parquet'):
+        """
+        Initialize the DataManager instance.
+
+        :param storage_type: Specifies the storage type to be used. Can be either 'in_memory' or 'filesystem'.
+        :param file_format: Specifies the file format to be used. Can be either 'parquet' or 'csv'.
+        """
+
+        if (storage_type not in self.VALID_STORAGE_TYPES):
+            storage_type = 'in_memory'
+            logging.warning(f"Invalid storage_type '{storage_type}' defaulting to 'in_memory', valid options are "
+                            f"{self.VALID_STORAGE_TYPES}")
+
+        if (file_format not in self.VALID_FILE_FORMATS):
+            file_format = 'parquet'
+            logging.warning(f"Invalid file_format '{file_format}' defaulting to 'parquet', valid options are "
+                            f"{self.VALID_FILE_FORMATS}")
+
+        self._file_format = file_format
+        self._fs = fsspec.filesystem('file')
+        self._manifest = {}
+        self._records = {}
+        self._storage_dir = None
+        self._storage_type = storage_type
+        self._total_rows = 0
+
+        if (storage_type == 'filesystem'):
+            self._storage_dir = tempfile.mkdtemp()
+
+        if (file_format == 'parquet'):
+            self._data_reader = cudf.read_parquet
+        elif (file_format == 'csv'):
+            self._data_reader = cudf.read_csv
+        else:
+            raise ValueError(f"Invalid file_format '{self._file_format}'")
+
+    def __contains__(self, item: Any) -> bool:
+        return item in self._records
+
+    def __del__(self):
+        if (self._storage_type == 'filesystem'):
+            shutil.rmtree(self._storage_dir)
+
+    def __len__(self) -> int:
+        return len(self._records)
+
+    def __repr__(self):
+        return (f"DataManager(records={self.num_rows}, "
+                f"storage_type={self._storage_type!r}, "
+                f"storage directory={self._storage_dir!r})")
+
+    def __str__(self):
+        return (f"DataManager with {self.num_rows} records, "
+                f"storage type: {self._storage_type}, "
+                f"storage directory: {self._storage_dir}")
+
+    def _update_manifest(self, source_id: uuid.UUID, action: str) -> None:
+        data_record = self._records[source_id]
+
+        if action == 'store':
+            self._manifest[source_id] = data_record.data
+            self._total_rows += data_record.num_rows
+        elif action == 'remove':
+            self._total_rows -= data_record.num_rows
+            del self._manifest[source_id]
```

Comment:
Why have a separate manifest? Wouldnt it be easier to generate this on demand from the `_records` when its requested? Same with `_total_rows`?